packages/toc/src/toc_item.tsx

```diff
@@ -2,7 +2,69 @@
 // Distributed under the terms of the Modified BSD License.
 
 import * as React from 'react';
-import { IHeading } from './utils/headings';
+import { IHeading, INotebookHeading } from './utils/headings';
+import { CommandRegistry } from '@lumino/commands';
+import { Menu } from '@lumino/widgets';
+import { CodeCell } from '@jupyterlab/cells';
+import { NotebookPanel } from '@jupyterlab/notebook';
+
+/**
+ * The command IDs used by TOC item.
+ */
+namespace CommandIDs {
+  export const runCells = 'toc:run-cells';
+}
+
+/**
+ * Tests whether a heading is a notebook heading.
+ *
+ * @private
+ * @param heading - heading to test
+ * @returns boolean indicating whether a heading is a notebook heading
+ */
+function isNotebookHeading(heading: any): heading is INotebookHeading {
+  return heading.type !== undefined && heading.cellRef !== undefined;
+}
+
+/**
+ * Checks whether a heading has runnable code cells.
+ *
+ * @private
+ * @param headings - list of headings
+ * @param heading - heading
+ * @returns boolean indicating whether a heading has runnable code cells
+ */
+function hasCodeCells(headings: IHeading[], heading: IHeading): boolean {
+  let h: INotebookHeading;
+  let i: number;
+
+  if (!isNotebookHeading(heading)) {
+    return false;
+  }
+  // Find the heading in the list of headings...
+  for (i = 0; i < headings.length; i++) {
+    if (heading === headings[i]) {
+      break;
+    }
+  }
```

Comment:
Would it make sense to use heading.indexOf here?

Reply:
made the change:)